cli/src/cli_util.rs

```diff
@@ -1304,10 +1303,23 @@ See https://github.com/martinvonz/jj/blob/main/docs/working-copy.md#stale-workin
         }
         self.user_repo = ReadonlyUserRepo::new(tx.commit(description));
         self.report_repo_changes(ui, &old_repo)?;
-
         if self.may_update_working_copy {
             if let Some(new_commit) = &maybe_new_wc_commit {
-                self.update_working_copy(ui, maybe_old_wc_commit.as_ref(), new_commit)?;
+                self.update_working_copy(ui, maybe_old_wc_commit.as_ref(), &new_commit)?;
+                println!("{:?}", new_commit.id());
+                if self.check_rewritable([new_commit.id()]).is_err() {
+                    let settings = self.settings.clone();
+                    let mut tx = self.start_transaction();
+                    let _top_commit = tx
+                        .mut_repo()
+                        .new_commit(
+                            &settings,
+                            vec![new_commit.id().clone()],
+                            new_commit.tree_id().clone(),
+                        )
+                        .write()?;
+                    tx.finish(ui, "added mutable commit to avoid @ immutable.")?;
+                }
```

Comment:
What I meant in Discord was, to move the check if the wc commit is immutable early into the function and then checking the bool if we should just write a new commit. If it is true then, you can finally print a message to `ui.status()` here.

Reply:
I don't get what would change if we check earlier in the function

Reply:
i don't get why checking earlier would change something, don't i still need to unwrap value of maybe_new_wc_commit to check if it's immutable or not ?

Reply:
which repo are you talking about ?  self.repo or tx.repo